From 7e65231abe2cf47e65b1c510bb61e416e4d2ff82 Mon Sep 17 00:00:00 2001
From: "Bobby R. Bruce" <bbruce@ucdavis.edu>
Date: Tue, 8 Mar 2022 12:53:35 -0800
Subject: [PATCH 385/757] util-docker: Adding docker-compose.yaml

"docker-compose.yaml" is used by the docker-compose utility to define
Docker image builds. In the case the "docker-compose.yaml" file defines
the building of all Docker images used in the gem5 project. These can be
built locally using `docker-compose build`.

Change-Id: I2544ec43e3a1743884e5aa243905704cb263d7d9
Reviewed-on: https://gem5-review.googlesource.com/c/public/gem5/+/57429
Reviewed-by: Jason Lowe-Power <power.jg@gmail.com>
Maintainer: Jason Lowe-Power <power.jg@gmail.com>
Tested-by: kokoro <noreply+kokoro@google.com>
---
 util/dockerfiles/README.md           |   3 +
 util/dockerfiles/docker-compose.yaml | 106 +++++++++++++++++++++++++++++++++++
 2 files changed, 109 insertions(+)
 create mode 100644 util/dockerfiles/README.md
 create mode 100644 util/dockerfiles/docker-compose.yaml

diff --git a/util/dockerfiles/README.md b/util/dockerfiles/README.md
new file mode 100644
index 0000000..18686fe
--- /dev/null
+++ b/util/dockerfiles/README.md
@@ -0,0 +1,3 @@
+This directory contains Dockerfiles used to create images used in the gem5 project.
+The `docker-compose.yaml` defines the building of each image.
+The images can be built locally using `docker-compose build`.
diff --git a/util/dockerfiles/docker-compose.yaml b/util/dockerfiles/docker-compose.yaml
new file mode 100644
index 0000000..035fea4
--- /dev/null
+++ b/util/dockerfiles/docker-compose.yaml
@@ -0,0 +1,106 @@
+version: '2'
+
+services:
+    gcn-gpu:
+        build:
+            context: gcn-gpu
+            dockerfile: Dockerfile
+        image: gcr.io/gem5-test/gcn-gpu
+    sst:
+        build:
+            context: sst-11.1.0
+            dockerfile: Dockerfile
+        image: gcr.io/gem5-test/sst-env
+    ubuntu-18.04_all-dependencies:
+        build:
+            context: ubuntu-18.04_all-dependencies
+            dockerfile: Dockerfile
+        image: gcr.io/gem5-test/ubuntu-18.04_all-dependencies
+    ubuntu-20.04_all-dependencies:
+        build:
+            context: ubuntu-20.04_all-dependencies
+            dockerfile: Dockerfile
+        image: gcr.io/gem5-test/ubuntu-20.04_all-dependencies
+    ubuntu-20.04_min-dependencies:
+        build:
+            context: ubuntu-20.04_min-dependencies
+            dockerfile: Dockerfile
+        image: gcr.io/gem5-test/ubuntu-20.04_min-dependencies
+    gcc-7:
+        build:
+            context: ubuntu-18.04_gcc-version
+            dockerfile: Dockerfile
+            args:
+                - version=7
+        image: gcr.io/gem5-test/gcc-version-7
+    gcc-8:
+        build:
+            context: ubuntu-18.04_gcc-version
+            dockerfile: Dockerfile
+            args:
+                - version=8
+        image: gcr.io/gem5-test/gcc-version-8
+    gcc-9:
+        build:
+            context: ubuntu-20.04_gcc-version
+            dockerfile: Dockerfile
+            args:
+                - version=9
+        image: gcr.io/gem5-test/gcc-version-9
+    gcc-10:
+        build:
+            context: ubuntu-20.04_gcc-version
+            dockerfile: Dockerfile
+            args:
+                - version=10
+        image: gcr.io/gem5-test/gcc-version-10
+    gcc-11:
+        build:
+            context: ubuntu-20.04_gcc-version-11
+            dockerfile: Dockerfile
+            args:
+                - version=11
+        image: gcr.io/gem5-test/gcc-version-11
+    clang-6:
+        build:
+            context: ubuntu-18.04_clang-version
+            dockerfile: Dockerfile
+            args:
+                - version=6.0
+        image: gcr.io/gem5-test/clang-version-6.0
+    clang-7:
+        build:
+            context: ubuntu-18.04_clang-version
+            dockerfile: Dockerfile
+            args:
+                - version=7
+        image: gcr.io/gem5-test/clang-version-7
+    clang-8:
+        build:
+            context: ubuntu-18.04_clang-version
+            dockerfile: Dockerfile
+            args:
+                - version=8
+        image: gcr.io/gem5-test/clang-version-8
+    clang-9:
+        build:
+            context: ubuntu-18.04_clang-version
+            dockerfile: Dockerfile
+            args:
+                - version=9
+        image: gcr.io/gem5-test/clang-version-9
+    clang-10:
+        build:
+            context: ubuntu-20.04_clang-version
+            dockerfile: Dockerfile
+            args:
+                - version=10
+        image: gcr.io/gem5-test/clang-version-10
+    clang-11:
+        build:
+            context: ubuntu-20.04_clang-version
+            dockerfile: Dockerfile
+            args:
+                - version=11
+        image: gcr.io/gem5-test/clang-version-11
+
-- 
1.8.3.1

